hwmv2: atmel_sam: Port to HWMv2

nandojve opened the pull request:
This ports all the Atmel SAM SoC's family and all related boards to HWMv2.

nandojve committed:
soc: atmel: Port SAM family to HWMv2

Port all the Atmel SAM SoCs to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: sam4e_xpro: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: arduino_due: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: sam4l_ek: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: sam4s_xplained: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: sam_e70_xplained: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: sam_v71_xult: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
boards: RoboKit1: Convert to HWMv2

Port the board to HWMv2.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
CODEOWNERS: Update Atmel entries

Update Atmel related rules to keep consistent.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
MAINTAINERS: Update Atmel entries

Update Atmel related rules to keep consistent.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nordicjm commented on soc/atmel/sam/Kconfig:
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) 2024 Gerson Fernando Budke <nandojve@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FAMILY_SAM
```
This Kconfig needs to match the name, the name is "atmel_sam" so needs to be `SOC_FAMILY_ATMEL_SAM`

nandojve replied:
Hi @nordicjm ,

This seems to be inconsistent in the tree then. This is one example https://github.com/zephyrproject-rtos/zephyr/blob/6acabd159aeeb644b8dc9d17012e14d25ce9684d/soc/st/stm32/Kconfig.soc#L6

We are using the same structure vendor/family. If I change this I need to update the hal for two different families SAM/SAM0.

Can we clarify this ?

nandojve replied:
I was wondering if it is not possible perform this change after the collab branch be merged in the main. Why? Then it is easy to perform the tree wide change since it requires hal change too. We can open an issue to address as a suggestion.

nandojve replied:
Hi @nordicjm ,

Did you had time to check this?

tejlmand replied:
it would be nice to get it aligned with: https://github.com/zephyrproject-rtos/zephyr/blob/c2d5d076c5d901188384cdfc5cbe5cfe3e292663/soc/atmel/sam/soc.yml#L2

especially as we plan for having compliance checks in place in future to ensure such names are aligned between Kconfig, CMake, yml-files in order to avoid the legacy mess of names being different across tools.

Could you place a followup comment https://github.com/zephyrproject-rtos/zephyr/issues/51831 under "Open issues" and place your name next to it 😄 

Then I'd be happy to accept this part of PR as-is.

You are of course also welcome to create a dedicated issue for tracking such work.

nandojve replied:
Added entry at #51831 and created the issue #69046 to easy track.

nordicjm commented on soc/atmel/sam/sam3x/Kconfig:
```diff
@@ -16,6 +15,3 @@ config SOC_SERIES_SAM3X
 	select PLATFORM_SPECIFIC_INIT
```
the family select goes in Kconfig.soc

nordicjm commented on soc/atmel/sam/sam3x/Kconfig.soc:
```diff
@@ -0,0 +1,42 @@
+
+config SOC_SERIES_SAM3X
+	bool
+	help
```
i.e. here

nandojve replied:
Ok, so the series now `selects` the family. Is it not a `depends on`, right?

nandojve replied:
Hi @nordicjm ,

Did you had time to check and confirm this?

tejlmand replied:
> Ok, so the series now `selects` the family. Is it not a `depends on`, right?

correct.

nordicjm commented on soc/atmel/sam/sam4e/Kconfig:
```diff
@@ -16,6 +15,3 @@ config SOC_SERIES_SAM4E
 	select PLATFORM_SPECIFIC_INIT
```
As above

nordicjm commented on soc/atmel/sam/sam4l/Kconfig:
```diff
@@ -0,0 +1,12 @@
+	select CPU_CORTEX_M4
+	select CPU_CORTEX_M_HAS_DWT
+	select CPU_HAS_ARM_MPU
+	select SOC_FAMILY_SAM
```
As above

nordicjm commented on soc/atmel/sam/sam4s/Kconfig:
```diff
@@ -15,7 +14,3 @@ config SOC_SERIES_SAM4S
 	select PLATFORM_SPECIFIC_INIT
```
As above, and for the rest

nordicjm commented on soc/atmel/sam/same70/Kconfig.soc:
```diff
@@ -0,0 +1,121 @@
+	default "same70n21b" if SOC_SAME70N21B
+	default "same70q19b" if SOC_SAME70Q19B
+	default "same70q20b" if SOC_SAME70Q20B
+	default "same70q21b" if SOC_SAME70Q21B
```
newline missing at end of file

nordicjm commented on boards/atmel/sam/sam4l_ek/sam4l_ek.dts:
```diff
@@ -17,6 +17,7 @@
 		i2c-0 = &twim0;
 		led0 = &user_led;
 		sw0 = &sw0_dfu;
+		zephyr,entropy = &trng;
```
this doesn't have any relation to hwmv2

nandojve replied:
Inconsistency, I'll drop it.

nordicjm commented on boards/atmel/sam/sam_e70_xplained/sam_e70_xplained_same70q21b.yaml:
```diff
@@ -1,22 +1,26 @@
-identifier: sam_e70b_xplained
+identifier: sam_e70_xplained_same70q21b
```
slashes in hwmv2: `identifier: sam_e70_xplained/same70q21b`

nordicjm commented on boards/atmel/sam/sam_v71_xult/sam_v71_xult_samv71q21.yaml:
```diff
@@ -1,27 +1,33 @@
-identifier: sam_v71_xult
+identifier: sam_v7b_xult_samv71q21
```
slashes

nordicjm commented on boards/atmel/sam/sam_v71_xult/sam_v7b_xult_samv71q21b.yaml:
```diff
@@ -1,27 +1,33 @@
-identifier: sam_v71b_xult
+identifier: sam_v7b_xult_samv71q21b
```
etc.

tejlmand commented on soc/soc_legacy/arm/atmel_sam/sam3x/Kconfig.soc:
```diff
@@ -1,89 +0,0 @@
-		bool "SAM3X4E"
-	config SOC_PART_NUMBER_SAM3X8C
-		bool "SAM3X8C"
-	config SOC_PART_NUMBER_SAM3X8E
```
the SoC part numbers seems to have gone missing.

It's used here:
https://github.com/zephyrproject-rtos/zephyr/blob/00deb8dfb10fa7da8dfcb3719fb93fe410924b68/samples/drivers/lcd_hd44780/src/main.c#L74

Please ensure the SoC part numbers are still available, as well as checking other settings from this files, as I believe they should be preserved.

nandojve replied:
I renamed as the below snip:

```diff
-config SOC_PART_NUMBER_SAME70Q21B
+config SOC_SAME70Q21B
	bool
	select SOC_SERIES_SAME70
	select SOC_ATMEL_SAME70_REVB

config SOC
	default "same70j19" if SOC_SAME70J19
```

tejlmand commented on boards/atmel/sam/sam_v71_xult/Kconfig.sam_v71_xult:
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2024 Gerson Fernando Budke <nandojve@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_SAM_V71_XULT_SAMV71Q21
+	bool
+	select SOC_SAMV71Q21
+
+config BOARD_SAM_V71_XULT_SAMV71Q21B
+	bool
+	select SOC_SAMV71Q21B
```
Please, no bool, as the `BOARD_<name>` and `BOARD_<full_identifier>` are constructed here:
https://github.com/zephyrproject-rtos/zephyr/blob/db742d03aa8fa469a96435fe85b292e62d061d38/boards/Kconfig.v2#L8
https://github.com/zephyrproject-rtos/zephyr/blob/db742d03aa8fa469a96435fe85b292e62d061d38/boards/Kconfig.v2#L13

Some details can be found here: https://docs.zephyrproject.org/latest/build/kconfig/tips.html#common-kconfig-shorthands

Please do:
```suggestion
config BOARD_SAM_V71_XULT
	select SOC_SAMV71Q21 if BOARD_SAM_V71_XULT_SAMV71Q21
	select SOC_SAMV71Q21B if BOARD_SAM_V71_XULT_SAMV71Q21B
```
as that will ensure any typos or similar wrong selection is caughtr by CI.

tejlmand reviewed (changes requested):
Looks good, but a little more polishing is needed.

Not far from done 😄

nandojve commented on soc/soc_legacy/arm/atmel_sam/samv71/Kconfig.soc:
```diff
@@ -1,168 +0,0 @@
-	  to achieve this, make CPU go to Wait mode instead of Sleep
-	  mode while using external crystal oscillator for main clock.
-
-config SOC_ATMEL_SAMV71_DISABLE_ERASE_PIN
```
common/Kconfig

nandojve committed:
tests: atmel_sam: Update platform name

This update all atmel_sam entries to use the new platform name related
to sam_e70_xplained and sam_v71_xult.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

nandojve committed:
samples: atmel_sam: Update platform name

This update all atmel_sam entries to use the new platform name related
to sam_e70_xplained and sam_v71_xult.

Signed-off-by: Gerson Fernando Budke <nandojve@gmail.com>

pdgendt commented on soc/atmel/sam/sam3x/soc.c:
I like the work done to have the same `Kconfig` symbols for clocks on all SAM SoCs, I nice next step would be to have a single `clock_init` function for all SoCs, or even have it moved to a driver.
